source/grid/manifold_lib.cc

```diff
@@ -986,8 +999,8 @@ namespace
                 weights[0] = 1. - line_point;
                 weights[1] = line_point;
                 new_point += my_weight *
-                             cell.line(line)->get_manifold().get_new_point(points_view,
-                                                                           weights_view);
+                             tria.get_manifold(line_manifold_id).get_new_point(points_view,
+                                 weights_view);
```

Comment:
I don't think the lookup into the map is terribly inefficient at this point - maybe now after my optimizations. I had initially thought that I might want to cache the manifold ids more directly somehow but then it never really showed up any more. And I don't think I will have missed it in callgrind since it is a function in `tria.cc` and thus not inlined.